src/common/EIP712/ForwarderUtil.ts

```diff
@@ -0,0 +1,23 @@
+import { GsnDomainSeparatorType, GsnRequestType } from './TypedRequestData'
+import { IForwarderInstance } from '../../../types/truffle-contracts'
+import { Contract } from 'web3-eth-contract'
+
+// register a forwarder for use with GSN: the request-type and domain separator we're using.
+export async function registerForwarderForGsn (forwarderTruffleOrWeb3: IForwarderInstance|Contract, sendOptions: any = undefined): Promise<void> {
```

Comment:
Please pick one or the other.  Or create an override for both. Please do not create a single method for **two** different kinds of contract object instance. This is what 'no'.

Reply:
why? yes, idealy, I'd have an overload with different types - but there are no overloads in javascript/typescript.

we moved to TruffleContract, but deployment still uses web3 contract, hence I need two flavours.
I think both should be removed in favor of "ethers", which is has API simpler to use than both, and more efficient resource-wise.

so I can either:
- add `registerForwarderForGsnWithTruffle` method for tests
- use `registerForwarderForGsn(forwarderContract.contract)` in all tests
- create a wrapper `registerForwarderForGsn` function, just for tests

do you think any of these is better?